Shared/API/Connection+Cloud.swift

```diff
@@ -54,39 +82,73 @@ struct GetAllConnections: CloudFunction {
                                 viewsToIgnore: viewsToIgnore).map { (value) -> [Connection] in
-                                    if let dict = value as? [String: [Connection]] {
-                                        var all: [Connection] = []
-
-                                        switch self.direction {
-                                        case .incoming:
-                                            if let incoming = dict["incoming"] {
-                                                all = incoming
-                                            }
-                                        case .outgoing:
-                                            if let outgoing = dict["outgoing"] {
-                                                all = outgoing
-                                            }
-                                        case .all:
-                                            if let incoming = dict["incoming"] {
-                                                all.append(contentsOf: incoming)
-                                            }
-                                            if let outgoing = dict["outgoing"] {
-                                                all.append(contentsOf: outgoing)
-                                            }
-                                        }
-
-                                        return all
-                                    } else {
-                                        return []
-                                    }
-                                }.eraseToAnyPublisher()
+            if let dict = value as? [String: [Connection]] {
+                var all: [Connection] = []
+
+                switch self.direction {
+                case .incoming:
+                    if let incoming = dict["incoming"] {
+                        all = incoming
+                    }
+                case .outgoing:
+                    if let outgoing = dict["outgoing"] {
+                        all = outgoing
+                    }
+                case .all:
+                    if let incoming = dict["incoming"] {
+                        all.append(contentsOf: incoming)
+                    }
+                    if let outgoing = dict["outgoing"] {
+                        all.append(contentsOf: outgoing)
+                    }
+                }
+
+                return all
+            } else {
+                return []
+            }
+        }.eraseToAnyPublisher()
     }
-}
-
 
+    func makeAsyncRequest(andUpdate statusables: [Statusable],
+                          viewsToIgnore: [UIView]) async throws -> [Connection] {
+
+        let result = try await self.makeAsyncRequest(andUpdate: statusables,
+                                                     params: [:],
+                                                     callName: "getConnections",
+                                                     viewsToIgnore: viewsToIgnore)
```

Comment:
A slightly more complex async conversion. The function suspends until we get a response. Once the response is received, we'll try to cast it to the proper type. If that fails we throw an error.